collectors_app.php

```diff
@@ -21,68 +21,108 @@
-$db = new PDO(
-    'mysql:host=DB;dbname=collection',
-    'root',
-    'password'
-);
-//----------------------------------------------------------------------------------------------------------------------defaulting
-$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
-//----------------------------------------------------------------------------------------------------------------------preparing query
-$query = $db->prepare('SELECT * FROM `socks`');
-//----------------------------------------------------------------------------------------------------------------------execute query
-$result = $query->execute();
+// validation of inputs:
+// DB table for sizes, color, pattern ✅
+// dropdowns that populate colors, sizes, patterns ✅
 
-if ($result) {
-    $socks = $query->fetchAll();//--------------------------------------------------------------------------------------REASSIGN PLAYERS ARRAY
-} else {
-    echo 'not working';
+//----------------------------------------------------------------------------------------------------------------------dropdowns
+//----------------------------------------------------------------------------------------------------------------------connect DB function
+function connectDB() {
+    $db = new PDO(
+        'mysql:host=DB;dbname=collection',
+        'root',
+        'password'
+    );
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    return $db;
 }
-//----------------------------------------------------------------------------------------------------------------------form $_GET data
-$test = '';
-if ($_GET) {
-    $test = "{$_GET['size']} {$_GET['pattern']} {$_GET['color']}";
-
-    //------------------------------------------------------------------------------------------------------------------move form to DB
-    $query = $db->prepare("INSERT INTO `socks` (`size`, `pattern`, `color`) VALUES (:size, :pattern, :color)");
-    $result = $query->execute([
-        'size' => $_GET['size'],
-        'pattern' => $_GET['pattern'],
-        'color' => $_GET['color']
-    ]);
+//----------------------------------------------------------------------------------------------------------------------get table function
+function getTable($table, $db) {
+    $query = $db->prepare("SELECT `{$table}` FROM `{$table}s`");
+    $result = $query->execute();
+    if ($result) {
+        return $query->fetchAll();
+    } else {
+        echo 'not working'; //maybe return not working
+    }
+}
+//----------------------------------------------------------------------------------------------------------------------creating html options for select dropdown
+function dropDownOptions($rows) {
+    $options= '';
+    foreach($rows as $row) {
+        foreach ($row as $key => $value) {
+            $options .= "<option value=\"{$value}\">{$value}</option>";
+        }
+    }
```

Comment:
Does this need a nested loop?

Reply:
I believe this loop is dealing with a larger array containing smaller arrays of each a single key value => pair. Is this to say because the inner array is a single pair, that I can do without the loop? Something like $row['key'] where 'key' is replaced by whatever $name was passed into this function?